.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -euo pipefail
+
+
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "elastic-agent-package" ]]; then
+  if [[ "$BUILDKITE_STEP_KEY" == "package_elastic-agent" ]]; then
+    export DOCKER_USERNAME_SECRET=$(vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
+    export DOCKER_PASSWORD_SECRET=$(vault kv get -field password "${DOCKER_REGISTRY_SECRET_PATH}")
+    docker login -u "${DOCKER_USERNAME_SECRET}" -p "${DOCKER_PASSWORD_SECRET}" "${DOCKER_REGISTRY}" 2>/dev/null
```

Comment:
I missed a docker logout command to ensure credentials are not left behind.

In fleet-server, it is used `pre-exit` hook for that:
https://github.com/elastic/fleet-server/blob/5091e66367835903bd07a4288be46c2d3b2ba986/.buildkite/hooks/pre-exit#L9

Reply:
in addition to the docker-logout, we should remove any authentication file that is left behind.